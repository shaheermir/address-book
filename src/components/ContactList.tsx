import React from 'react'
import { TContact } from '../features/types'
import ContactCard from './ContactCard'

type TProps = {
  contacts: TContact[]
}
function ContactList({ contacts }: TProps) {
  return (
    <div>
      {contacts.map((contact) => (
        <ContactCard key={contact.id} contact={contact} />
      ))}
    </div>
  )
}

export default ContactList
