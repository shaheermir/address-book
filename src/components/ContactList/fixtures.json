[
  {
    "name": {
      "title": "Mr",
      "first": "Allan",
      "last": "Nguyen"
    },
    "email": "allan.nguyen@example.com",
    "phone": "013873 67584",
    "picture": {
      "large": "https://randomuser.me/api/portraits/men/40.jpg",
      "medium": "https://randomuser.me/api/portraits/med/men/40.jpg",
      "thumbnail": "https://randomuser.me/api/portraits/thumb/men/40.jpg"
    },
    "id": "0-allan-nguyen"
  },
  {
    "name": {
      "title": "Mr",
      "first": "Connor",
      "last": "Mitchell"
    },
    "email": "connor.mitchell@example.com",
    "phone": "(799)-330-4934",
    "picture": {
      "large": "https://randomuser.me/api/portraits/men/1.jpg",
      "medium": "https://randomuser.me/api/portraits/med/men/1.jpg",
      "thumbnail": "https://randomuser.me/api/portraits/thumb/men/1.jpg"
    },
    "id": "1-connor-mitchell"
  },
  {
    "name": {
      "title": "Mr",
      "first": "Ricky",
      "last": "Pearson"
    },
    "email": "ricky.pearson@example.com",
    "phone": "(754)-624-7123",
    "picture": {
      "large": "https://randomuser.me/api/portraits/men/74.jpg",
      "medium": "https://randomuser.me/api/portraits/med/men/74.jpg",
      "thumbnail": "https://randomuser.me/api/portraits/thumb/men/74.jpg"
    },
    "id": "2-ricky-pearson"
  }
]
