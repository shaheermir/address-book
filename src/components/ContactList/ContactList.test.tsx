import React from 'react'
import { shallow } from 'enzyme'
import { ContactList } from './ContactList'
import contacts from './fixtures.json'

describe('ContactList', () => {
  it('renders contact list', () => {
    const link = shallow(<ContactList contacts={contacts} />)
    expect(link).toMatchSnapshot()
  })
})
