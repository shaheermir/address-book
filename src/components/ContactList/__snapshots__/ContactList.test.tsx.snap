// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList renders contact list 1`] = `
<styled.div>
  <ContactCard
    contact={
      Object {
        "email": "allan.nguyen@example.com",
        "id": "0-allan-nguyen",
        "name": Object {
          "first": "Allan",
          "last": "Nguyen",
          "title": "Mr",
        },
        "phone": "013873 67584",
        "picture": Object {
          "large": "https://randomuser.me/api/portraits/men/40.jpg",
          "medium": "https://randomuser.me/api/portraits/med/men/40.jpg",
          "thumbnail": "https://randomuser.me/api/portraits/thumb/men/40.jpg",
        },
      }
    }
    key="0-allan-nguyen"
  />
  <ContactCard
    contact={
      Object {
        "email": "connor.mitchell@example.com",
        "id": "1-connor-mitchell",
        "name": Object {
          "first": "Connor",
          "last": "Mitchell",
          "title": "Mr",
        },
        "phone": "(799)-330-4934",
        "picture": Object {
          "large": "https://randomuser.me/api/portraits/men/1.jpg",
          "medium": "https://randomuser.me/api/portraits/med/men/1.jpg",
          "thumbnail": "https://randomuser.me/api/portraits/thumb/men/1.jpg",
        },
      }
    }
    key="1-connor-mitchell"
  />
  <ContactCard
    contact={
      Object {
        "email": "ricky.pearson@example.com",
        "id": "2-ricky-pearson",
        "name": Object {
          "first": "Ricky",
          "last": "Pearson",
          "title": "Mr",
        },
        "phone": "(754)-624-7123",
        "picture": Object {
          "large": "https://randomuser.me/api/portraits/men/74.jpg",
          "medium": "https://randomuser.me/api/portraits/med/men/74.jpg",
          "thumbnail": "https://randomuser.me/api/portraits/thumb/men/74.jpg",
        },
      }
    }
    key="2-ricky-pearson"
  />
</styled.div>
`;
